internal/app/wwctl/overlay/chown/main.go

```diff
@@ -11,36 +11,43 @@ import (
 	"github.com/hpcng/warewulf/internal/pkg/util"
 	"github.com/hpcng/warewulf/internal/pkg/wwlog"
 
+	"github.com/pkg/errors"
 	"github.com/spf13/cobra"
 )
 
 func CobraRunE(cmd *cobra.Command, args []string) error {
 	var overlaySourceDir string
-	overlayName := args[0]
-	fileName := args[1]
 	var uid int
 	var gid int
 	var err error
 
-	uid, err = strconv.Atoi(args[2])
+	overlayKind := args[0]
+	overlayName := args[1]
+	fileName := args[2]
+
+	if overlayKind != "system" && overlayKind != "runtime" {
+		return errors.New("overlay kind must be of type 'system' or 'runtime'")
+	}
+
+	uid, err = strconv.Atoi(args[3])
 	if err != nil {
-		wwlog.Printf(wwlog.ERROR, "UID is not an integer: %s\n", args[2])
+		wwlog.Printf(wwlog.ERROR, "UID is not an integer: %s\n", args[3])
 		os.Exit(1)
 	}
 
 	if len(args) > 3 {
```

Comment:
Missed increasing the len of arguments to 4 here.

Reply:
Good catch, fixed!